testing/src/test/scala/sttp/tapir/testing/EndpointVerifierTest.scala

```diff
@@ -286,4 +289,30 @@ class EndpointVerifierTest extends AnyFlatSpecLike with Matchers {
 
     result shouldBe Set()
   }
+
+  it should "detect endpoints with body where status code doesn't allow a body" in {
+
+    val e1 = endpoint.in("endpoint1_Err").out(stringBody).out(statusCode(StatusCode.NoContent))
+    val e2 = endpoint.in("endpoint2_Ok").out(stringBody).out(statusCode(StatusCode.BadRequest))
+    val e3 = endpoint.in("endpoint3_Err").out(stringBody).out(statusCode(StatusCode.NotModified))
+    val e4 = endpoint.in("endpoint4_ok").out(emptyOutputAs(NoContent)).out(statusCode(StatusCode.NoContent))
+    val e5 = endpoint
```

Comment:
Would the checker pass for an endpoint such as:

```scala
val e6 = endpoint
      .in("endpoint6_ok")
       .errorOut(
         sttp.tapir.oneOf[ErrorInfo](
           oneOfVariant(statusCode(StatusCode.NotFound).and(jsonBody[NotFound])),
           oneOfVariant(statusCode(StatusCode.NoContent).and(emptyOutputAs[NoContent]))
         )
       )
```

Reply:
Good idea to check this, I have added a test.